From 9bb5ded1096dd41e13bc21e1f516e24c2075d4da Mon Sep 17 00:00:00 2001
From: Rene Rebe <rene@exactcode.de>
Date: Sun, 2 Aug 2026 04:07:52 +0800
Subject: [PATCH 349/440] T2SDE: arch: x86: expand whitelist for no-speculation
 whitelist on 5x86 systems
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

AFAIC the upstream cpu vuln whitelist misses many 5x86 clones,
"CyrixInstead", "NexGenDriven", "RiseRiseRise", "SiS SiS SiS ",
"UMC UMC UMC ", "VIA VIA VIA ".

   - René Rebe

Link: https://github.com/rxrbln/t2sde/blob/aefea81ae35770a36de34f2ef969fac8fbb9f6fa/package/kernel/linux/5x86-no-speculation.patch
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 arch/x86/kernel/cpu/common.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/arch/x86/kernel/cpu/common.c b/arch/x86/kernel/cpu/common.c
index 558a253c92fac..1de1fc4fce1aa 100644
--- a/arch/x86/kernel/cpu/common.c
+++ b/arch/x86/kernel/cpu/common.c
@@ -1183,10 +1183,7 @@ static void identify_cpu_without_cpuid(struct cpuinfo_x86 *c)
 
 static const __initconst struct x86_cpu_id cpu_vuln_whitelist[] = {
 	VULNWL(ANY,	4, X86_MODEL_ANY,	NO_SPECULATION),
-	VULNWL(CENTAUR,	5, X86_MODEL_ANY,	NO_SPECULATION),
-	VULNWL(INTEL,	5, X86_MODEL_ANY,	NO_SPECULATION),
-	VULNWL(NSC,	5, X86_MODEL_ANY,	NO_SPECULATION),
-	VULNWL(VORTEX,	5, X86_MODEL_ANY,	NO_SPECULATION),
+	VULNWL(ANY,	5, X86_MODEL_ANY,	NO_SPECULATION),
 	VULNWL(VORTEX,	6, X86_MODEL_ANY,	NO_SPECULATION),
 
 	/* Intel Family 6 */
-- 
2.55.0

